### PR TITLE
feat(sanity): adding CSI Sanity test

### PR DESCRIPTION
**Pull request by pawanpraka1**
Signed-off-by: Pawan <pawan@mayadata.io>

**What this PR does?**:

adding CSI sanity test for ZFS-LocalPV

**Changes?**:

1. convert volume name and snapshot name to lowercase as they are needed to create a CR and uppercase characters are not allowed
2.  returning proper error code in the code
3. added "wait" functionality to wait for volume/operation creation, needed since sanity does not have retry logic
4. add functionality to pass "node" where PV should be created, needed since sanity does not pass topology information
5. added few checks to return correct error codes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,6 +235,12 @@ zfs-driver-image: zfs-driver
 ci:
 	@echo "--> Running ci test";
 	$(PWD)/ci/ci-test.sh
+
+.PHONY: sanity
+sanity:
+	@echo "--> Running ci test";
```
Comment (kmova) [inline Makefile:241]: n_: running csi-sanity tests

File: ci/ci-test.sh [hunk 2]
```diff
@@ -89,7 +89,7 @@ set +e
 
 echo "running ginkgo test case"
 
-ginkgo -v
+# ginkgo -v
```
Comment (kmova) [inline ci/ci-test.sh:92]: n_ remove it.

File: pkg/driver/controller.go [hunk 3]
```diff
@@ -126,17 +159,39 @@ func CreateZFSVolume(req *csi.CreateVolumeRequest) (string, error) {
...
+			}
+		} else {
+			if vol.Spec.Capacity != capacity {
+				return "", status.Errorf(codes.AlreadyExists,
+					"volume %s already present", volName)
+			}
+			return vol.Spec.OwnerNodeID, nil
+		}
+	}
+
```
Comment (payes) [inline pkg/driver/controller.go:179]: Should we add a return for errors other than NotFound.

Comment (pawanpraka1) [reply pkg/driver/controller.go:179]: this is Create request. We don't need to return any other error as we are creating the volume. The above check is added for sanity to check if volume is already present with the same capacity, then return successful response.

**Review by payes — APPROVED** [review]

File: pkg/driver/controller.go [hunk 4]
```diff
@@ -169,12 +224,18 @@ func CreateZFSVolume(req *csi.CreateVolumeRequest) (string, error) {
...
 	return selected, nil
 }
 
 // CreateVolClone creates the clone from a volume
 func CreateVolClone(req *csi.CreateVolumeRequest, srcVol string) (string, error) {
-	volName := req.GetName()
+	volName := strings.ToLower(req.GetName())
```
Comment (kmova) [inline pkg/driver/controller.go:238]: With this change are there any backward compatiblity check that need to be verified. Like, creating a snapshot or clone in older release and after upgrading trying to create new snapshot or delete old snap or clone. 

If the ids were always - pvc-uuid.. may be it is not an issue. just checking here.

Comment (pawanpraka1) [reply pkg/driver/controller.go:238]: Yes, it was always pvc-uuid.

**Review by kmova — APPROVED** [review]

**Pull request merged**